pkg/insights/insightsclient/insightsclient.go

```diff
@@ -162,7 +162,12 @@ func userAgent(releaseVersionEnv string, v apimachineryversion.Info, cv *configv
 	return fmt.Sprintf("insights-operator/%s cluster/%s", gitVersion, cv.Spec.ClusterID)
 }
 
-func (c *Client) getClusterVersion() (*configv1.ClusterVersion, error) {
+func (c *Client) SetClusterVersion(cv *configv1.ClusterVersion) *Client {
+	c.clusterVersion = cv
+	return c
+}
```

Comment:
This function is used only in the tests, right? If so then we should be able to mock it and remove this function IMO.

Reply:
Perhaps we could introduce some similar interface as in https://github.com/openshift/insights-operator/blob/master/pkg/gatherers/conditional/conditional_gatherer.go#L43 and then mock it?